contracts/src/interfaces/IHyperdriveRegistry.sol

```diff
@@ -2,10 +2,58 @@
 pragma solidity 0.8.20;
 
 interface IHyperdriveRegistry {
-    /// @notice Allows anyone to get the info for a hyperdrive instance.
-    /// @param _hyperdriveInstance The hyperdrive instance address.
-    /// @return The uint256 value set by governance.
+    /// @dev The info related to each Hyperdrive instance.
+    struct HyperdriveInfo {
+        /// @dev Data about the instance. Different registries can utilize
+        ///      different schemas for these values.
+        uint256 data;
+        /// @dev The factory that deployed this instance.
+        address factory;
+        /// @dev The base token of the instance.
+        IERC20 baseToken;
+        /// @dev The vault shares token of the instance.
+        IERC20 vaultSharesToken;
+        /// @dev The name of the instance.
+        string name;
+        /// @dev The version of the instance.
+        string version;
+    }
+
+    /// @notice Gets the number of Hyperdrive instances that have been registered.
+    /// @return The number of registered instances.
+    function getNumberOfHyperdriveInstances() external view returns (uint256);
```

Comment:
*pedantic:* feel like `count()` would work equally well here.

Reply:
I'm keeping this consistent with the `HyperdriveFactory`.